client/signup/steps/plans/index.jsx

```diff
@@ -82,10 +84,27 @@ export class PlansStep extends Component {
 			...additionalStepData,
 		};
 
-		this.props.submitSignupStep( step, {
-			cartItem,
-		} );
-		this.props.goToNextStep();
+		if ( flowName === 'site-selected' && ! cartItem ) {
+			wp.req
+				.post( {
+					path: `/domains/${ this.props.selectedSite.ID }/${ this.props.selectedSite.name }/convert-domain-only-to-site`,
+					apiVersion: '1.1',
+				} )
+				.catch( ( error ) => {
+					this.props.errorNotice( error.message );
+				} )
+				.finally( () => {
+					this.props.submitSignupStep( step, {
+						cartItem,
+					} );
+					this.props.goToNextStep();
+				} );
```

Comment:
I believe the `finally` clause runs even if there's an error, is that expected? If not, I think you can use the other call style, something like:

```
wp.req.post(
	`/domains/${ this.props.selectedSite.ID }/${ this.props.selectedSite.name }/convert-domain-only-to-site`,
	{},
	( error, data ) => {
		if ( error ) {
			this.props.errorNotice( error.message );
			return;
		}
		this.props.submitSignupStep( step, {
			cartItem,
		} );
		this.props.goToNextStep();
	}
);
```

Reply:
It was intended but you are right, it's better for the user stay in the same page in case of error. Thanks for the catch! :)